app/screens/channel/channel_title/index.tsx

```diff
@@ -0,0 +1,279 @@
+import CompassIcon from '@components/compass_icon';
+import FormattedText from '@components/formatted_text';
+import {General} from '@constants';
+import {MM_TABLES} from '@constants/database';
+import {useTheme} from '@context/theme';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {t} from '@utils/i18n';
+
+import type {Database} from '@nozbe/watermelondb';
+
+import type ChannelInfoModel from '@typings/database/models/servers/channel_info';
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type MyChannelSettingsModel from '@typings/database/models/servers/my_channel_settings';
+import type UserModel from '@typings/database/models/servers/user';
+
+const ConnectedChannelTitle = ({channel, onPress, canHaveSubtitle, currentUserId, channelInfo, channelSettings, teammateId, teammate}: ChannelTitleProps) => {
+    const theme = useTheme();
+
+    const style = getStyle(theme);
+    const channelType = channel.type;
+    const currentChannelName = channel ? channel.displayName : '';
+    const displayName = channel.displayName;
+    const hasGuests = channelInfo[0]?.guestCount > 0;
+    const isArchived = channel.deleteAt !== 0;
+    const isChannelMuted = channelSettings[0]?.notifyProps?.mark_unread === 'mention';
+    const isChannelShared = false; // todo: Read from ChannelModel when implemented
+
+    let isGuest = false;
+    let isSelfDMChannel = false;
+    let wrapperWidth = 90;
+
+    if (channel.type === General.DM_CHANNEL && teammate) {
+        //fixme: use util method from Elias' PR.
+        isGuest = false; //isTeamMateGuest(teammate.roles);
+        isSelfDMChannel = currentUserId === teammateId;
+    }
+
+    const renderArchiveIcon = () => {
+        let content = null;
+        if (isArchived) {
+            content = (
+                <CompassIcon
+                    name='archive-outline'
+                    style={[style.archiveIcon]}
+                />
+            );
+        }
+        return content;
+    };
+
+    const renderHasGuestsText = () => {
+        if (!canHaveSubtitle) {
+            return null;
+        }
+        if (!isGuest && !hasGuests) {
+            return null;
+        }
+        if (channelType === General.DM_CHANNEL && !isGuest) {
+            return null;
+        }
+
+        let messageId;
+        let defaultMessage;
+        if (channelType === General.DM_CHANNEL) {
+            messageId = t('channel.isGuest');
+            defaultMessage = 'This person is a guest';
+        } else if (channelType === General.GM_CHANNEL) {
+            messageId = t('channel.hasGuests');
+            defaultMessage = 'This group message has guests';
+        } else if (
+            channelType === General.OPEN_CHANNEL ||
+            channelType === General.PRIVATE_CHANNEL
+        ) {
+            messageId = t('channel.channelHasGuests');
+            defaultMessage = 'This channel has guests';
+        } else {
+            return null;
```

Comment:
when will this be reached? I think the above if statements are covering all cases

Reply:
They are covering all cases and if a different value for the channelType is passed, it will return null instead of the formatted text.